### PR TITLE
Resolve one many

### PR DESCRIPTION
**Pull request by SilverSixx**

### CODE REVIEW

File: src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java
Comment (suyttthideptrai) [inline src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java:1]: Thêm doc giải thích provider, consumer đóng vai trò là gì trong bài toán 1 nhiều

Comment (SilverSixx) [reply src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java:1]: oke

File: src/main/java/com/example/SS2_Backend/model/onetomany/Individual.java
Comment (han8129) [inline src/main/java/com/example/SS2_Backend/model/onetomany/Individual.java:1]: pankage naming should follow camel case for consistency

File: src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+    String fitnessFunction;
+    boolean providerFunctionStatus = false;
+    boolean consumerFunctionStatus = false;
+    boolean fitnessFunctionStatus = false;
+
+    public void setPopulation(ArrayList<Individual> individuals, String[] propertiesNames, int [][] excludedPairs) {
+        this.individuals = new IndividualList(individuals, propertiesNames);
+        this.excludedPairs = excludedPairs;
+        initializeFields();
+    }
```
Comment (han8129) [inline src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java:42]: should consider passing this in the *Constructor*

File: src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java
Comment (han8129) [inline src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java:1]: Problem initialization is more complex than it should be. `setPopulation()` and
`initializeFields()` can be replace with a simple constructor with some
parameters

`oneToManyMatching()`, `defaultFitnessEvaluation()` should be considered to be
placed in another file as it purpose is directly linked to the Problem class,
the Problem is just to hold the objectives, individuals, constraints, and
solutions. Move these utilities to another file should also make maintaining
easier

**Review by han8129 — CHANGES_REQUESTED** [review]
package name should be in camel case, and a Dto for decoding request's body is currently missing

**Review by suyttthideptrai — APPROVED** [review]

File: src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java (hunk 1 above)
Comment (suyttthideptrai) [reply src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java]: Tạm thời merge luôn, sửa đổi vào pull request khác

Comment (suyttthideptrai) [reply src/main/java/com/example/SS2_Backend/model/onetomany/MatchingProblem.java]: Nhắc Đạt chú ý đặt tên branch cho đúng rule

**Pull request merged**